components/ledger/ledger-utxo-flow/src/test/kotlin/net/corda/ledger/utxo/flow/impl/flows/backchain/TransactionBackchainResolutionFlowVersionedFlowFactoryTest.kt

```diff
@@ -22,8 +23,8 @@ class TransactionBackchainResolutionFlowVersionedFlowFactoryTest {
     }
 
     @Test
-    fun `with platform version 50099 creates TransactionBackchainResolutionFlowV1`() {
-        val flow = factory.create(50099, mock())
+    fun `with last 5_0 platform version creates TransactionBackchainResolutionFlowV1`() {
+        val flow = factory.create(CORDA_5_1.value - 1, mock())
```

Comment:
```suggestion
        val flow = factory.create(CORDA_5_0.value, mock())
```

This one is a bit more questionable than the earlier comments, because this would always test 5.0, not 5.0.1 for examples. So this suggestion may not be the best approach.

That said, I think the current implementation is not testing a "real" Corda version anyway; isn't it getting the _literal_ minus one value, i.e. 50099, whereas you may want the latest valid value, i.e. we should be getting the ordinal - 1 entry. Not sure how much this matters though.

Reply:
We know this flow upgrade should happen from 5.1, so anything before that should use V1.
Regardless of whether it exists or not (yet).
Also, the current test set checks the ends of the intervals, so it includes all possible values, not just the real ones.
This makes it a bit more future-proof and less likely to need to be adjusted when new versions get released.

Reply:
Would renaming the test to
`with last potential 5_0 ...`
make this intent cleaner?

Reply:
> Would renaming the test to `with last potential 5_0 ...` make this intent cleaner?

Yes, that sounds like a good compromise.